misk-hibernate/src/main/kotlin/misk/hibernate/ReflectionQueryFactory.kt

```diff
@@ -528,11 +527,37 @@ internal class ReflectionQuery<T : DbEntity<T>>(
       }
     }.toTypedArray())
 
-    private fun toValue(row: Any): Any? {
+    private fun toValue(row: Any?): Any? {
       return when {
+        row == null -> null
         constructor == null -> row
         properties.size == 1 -> constructor.call(row)
-        else -> constructor.call(*(row as Array<*>))
+        else -> {
+          val args = row as Array<*>
+          // If we got back an array of null values, which can't be assigned to the constructor,
+          // then we should return null.
+          if (args.all { it == null } && !constructor.parameters.all { it.type.isMarkedNullable } ) return null
+          // Otherwise, we should check if we *can* assign the null values to the constructor.
+          checkNullability(constructor, args)
+          // Now it's (probably!) safe to call the constructor.
+          constructor.call(*args)
+        }
+      }
+    }
+
+    private fun checkNullability(constructor: KFunction<*>, args: Array<*>) {
```

Comment:
Finding a way to remember this information would be really nice. This isn’t a particularly efficient operation and doing it once per row is bad.

https://publicobject.com/2016/03/24/reflection-machines/

Reply:
Thanks for pointing out the performance impact here! I will give this some thought!

Strictly speaking, I don't actually think we'll run into this very often (maybe only if the mysql server is running in a weird mode and someone forgets to group their aggegration?). 

I could just remove this and let an NPE occur in the rare instance that this happens.